q2_diversity_lib/plugin_setup.py

```diff
@@ -76,36 +78,170 @@
+                                   "for Pielou's Evenness."},
+    name="Pielou's Evenness",
+    description="Compute Pielou's Evenness for each sample in a "
+                "feature table",
     citations=[citations['pielou1966measurement']]
 )
 
+# TODO: Augment citations as needed
 plugin.methods.register_function(
     function=q2_diversity_lib.shannon_entropy,
     inputs={'table': FeatureTable[Frequency | RelativeFrequency]},
     parameters={'drop_undefined_samples': Bool},
     outputs=[('vector',
              SampleData[AlphaDiversity])],
-    input_descriptions={'table': 'The feature table containing the samples '
-                        'for which Shannon\'s Entropy should be computed.'},
-    parameter_descriptions={'drop_undefined_samples': 'Samples with no '
-                            'observed features produce undefined (NaN) values.'
-                            ' If true, these samples are dropped from the '
-                            'output vector.'},
-    output_descriptions={'vector': 'Vector containing per-sample values '
-                                   'for Shannon\'s Entropy.'},
-    name='Shannon\'s Entropy',
-    description='Compute Shannon\'s Entropy for each sample in a '
-                'feature table',
+    input_descriptions={'table': "The feature table containing the samples "
+                        "for which Shannon's Entropy should be computed."},
+    parameter_descriptions={'drop_undefined_samples': "Samples with no "
+                            "observed features produce undefined (NaN) values."
+                            " If true, these samples are dropped from the "
+                            "output vector."},
+    output_descriptions={'vector': "Vector containing per-sample values "
+                                   "for Shannon's Entropy."},
+    name="Shannon's Entropy",
+    description="Compute Shannon's Entropy for each sample in a "
+                "feature table",
     citations=[citations['shannon1948communication']]
 )
+
+# ------------------------ beta-diversity -----------------------
+# TODO: Augment citations as needed
+plugin.methods.register_function(
+    function=q2_diversity_lib.bray_curtis,
+    inputs={'table': FeatureTable[Frequency]},
+    parameters={'n_jobs': Int},
+    outputs=[('distance_matrix', DistanceMatrix)],
+    input_descriptions={
+        'table': "The feature table containing the samples for which "
+                 "Bray-Curtis dissimilarity should be computed."},
+    parameter_descriptions={
+        'n_jobs': "The number of CPU threads to use in performing this "
+                  "calculation.  More threads = faster performance. May not "
+                  "exceed the number of available physical cores. If n-jobs = "
+                  "-1, all CPUs are used. For n-jobs < -1, (n_cpus + 1 + "
+                  "n-jobs) are used. E.g if n-jobs = -2, all CPUs but"
+                  " one are used."},
```

Comment:
- `n_jobs == -1` is an anti-pattern in QIIME 2 - we have moved on to using signature's like:

```python
Int % Range(1, None) | Str % Choices(['auto'])
```

- Should this be `n_threads`, since we are dealing with CPU threads, and not jobs?
- Is the n_jobs < -1 business inherited from `skbio`, or is that something you've created here?
- For referencing the parameter descriptions inline, we try to avoid interface-specific representations, when possible. With that in mind, its our convention to default to "Artifact API" representations of param names, since that is more portable. Please change `n-jobs` to `n_jobs` (or `n_threads`, depending on the resolution to my question above).

Reply:
The `n_jobs` business is inherited from `sklearn` c/o `skbio`, and is currently supported in `q2_diversity`. It took me 5 minutes just to figure out how the math worked when I wrote this up, and I'm totally fine with restricting users to `Int % Range(1, None) | Str % Choices(['auto'])` if that's our preferred approach. 

Re: `n_threads` vs `n_jobs`, I'm inclined to stick with `n_jobs` for now, simply for reasons of consistency. `unifrac` uses `threads`, but most of our other measures are coming from `sklearn` c/o `skbio` which uses `jobs`. If you feel strongly about `threads` I'd be happy to change everything over, but I'm not inclined to support multiple names. Users get negative value out of having different names for params with equivalent purposes across methods.

Reply:
> and I'm totally fine with restricting users to Int % Range(1, None) | Str % Choices(['auto']) if that's our preferred approach.

Luckily the two aren't mutually exclusive! Maybe one option is to add a few Str Choices for the more common "negative value" cases - 'all-but-one', 'all-but-two', etc. Otherwise, I would be fine with dropping the negative-value support, its pretty niche (IMO).

> I'm inclined to stick with n_jobs for now

That works for me.

> Users get negative value out of having different names for params with equivalent purposes across methods.

I agree - which was why I brought it up in the first place: your help text is using two different names _within a single method_:

> 'n_**jobs**': "The number of **CPU threads** to use in performing this "

I assumed you're dealing with threads (since you wrote about them in the description text), which is why I asked you to change the _name_ of the param, but cross-referencing with sklearn, I think you actually need to be using the word "jobs" in the description. Make sense?

Reply:
> If you feel strongly about threads

The only thing that I feel strongly about is consistency, and using the right label for the method of parallelism (jobs are not the same thing as threads). We should be using whatever name or label the source implementations use - unless the source implementation is clearly incorrect.

Reply:
All clear. 
- Dropping support for `all-but-one`s for now. Can add later if requested, but feels over-engineered for most users. 
- I will absolutely use "jobs" in sklearn help text, but I am inclined to mix `n_jobs` param name with "number of threads" for `unifrac` metrics, which are actually using threads. Users get a consistent interface, and anyone who actually cares get accurate information. Is this a terrible idea?

Reply:
> Is this a terrible idea?

I'm not sure if its terrible, but I don't think its a great idea. The `unifrac` package works with threads (and refers to them as such) - I think you should match that in the unifrac methods.

> Users get a consistent interface

I think optimizing for the same "name" in the q2-div-lib beta methods is a noble idea, but I think it obfuscates the meaning of the upstream toolings' parallelism models. One way this might break down: cluster submission tools jobs vs threads _matter_, and if you're calling a thread a job (or vice versa), I'm sure someone will get sad about that someday.

In general we have found it is best to try to match parameter names to the upstream tool's name when wrapping in q2 plugins. Sometimes its better to use the "long-form" name, in the case of many CLI-tools (like cutadapt), but then the "lingo" is the same between plugin and upstream tool, which makes it really convenient for pushing support upstream (a common line I like to use w.r.t. q2-cutadapt is "please see the cutadapt docs for more detail - the parameters are the same between the plugin and the `cutadapt` tool").